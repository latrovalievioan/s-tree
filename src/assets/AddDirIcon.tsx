import { SVGProps } from 'react';

export const AddDirIcon = (props: SVGProps<SVGSVGElement>) => (
  <svg
    className="w-[20px] h-[20px] text-gray-800 dark:text-white"
    aria-hidden="true"
    xmlns="http://www.w3.org/2000/svg"
    fill="none"
    viewBox="0 0 18 18"
    width={32}
    height={32}
    {...props}
  >
    <path
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      d="M7 11h4m-2 2V9M2 5h14a1 1 0 0 1 1 1v10a1 1 0 0 1-1 1H2a1 1 0 0 1-1-1V6a1 1 0 0 1 1-1Zm5.443-4H2a1 1 0 0 0-1 1v3h9.943l-2.7-3.6a1 1 0 0 0-.8-.4Z"
    />
  </svg>
);
