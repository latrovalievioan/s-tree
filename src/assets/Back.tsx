import { SVGProps } from 'react';

export const Back = (props: SVGProps<SVGSVGElement>) => (
  <svg
    width={40}
    height={40}
    viewBox="0 0 80 80"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    {...props}
  >
    <circle cx={40} cy={40} r={36} stroke="currentColor" strokeWidth={8} />
    <path
      d="M45.333 23 28 40.2l17.333 17.2"
      stroke="currentColor"
      strokeWidth={8}
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
);
