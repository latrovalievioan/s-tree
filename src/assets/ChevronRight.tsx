import { SVGProps } from 'react';

export const ChevronRight = (props: SVGProps<SVGSVGElement>) => (
  <svg
    aria-hidden="true"
    xmlns="http://www.w3.org/2000/svg"
    fill="none"
    width={10}
    height={10}
    viewBox="0 0 6 10"
    {...props}
  >
    <path
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      d="m1 9 4-4-4-4"
    />
  </svg>
);
