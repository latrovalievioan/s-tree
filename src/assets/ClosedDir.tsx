import { SVGProps } from 'react';

export const ClosedDir = (props: SVGProps<SVGSVGElement>) => (
  <svg
    aria-hidden="true"
    xmlns="http://www.w3.org/2000/svg"
    fill="none"
    width={21}
    height={18}
    viewBox="0 0 18 18"
    {...props}
  >
    <path
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      d="M1 5v11a1 1 0 0 0 1 1h14a1 1 0 0 0 1-1V6a1 1 0 0 0-1-1H1Zm0 0V2a1 1 0 0 1 1-1h5.443a1 1 0 0 1 .8.4l2.7 3.6H1Z"
    />
  </svg>
);
