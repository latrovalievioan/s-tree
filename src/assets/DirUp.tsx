import { SVGProps } from 'react';

export const DirUp = (props: SVGProps<SVGSVGElement>) => (
  <svg
    width={24}
    height={24}
    viewBox="0 0 18 18"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    {...props}
  >
    <g
      clipPath="url(#a)"
      stroke="currentColor"
      strokeWidth={2}
      strokeLinecap="round"
      strokeLinejoin="round"
    >
      <path d="M1 5v11a1 1 0 0 0 1 1h14a1 1 0 0 0 1-1V6a1 1 0 0 0-1-1H1Zm0 0V2a1 1 0 0 1 1-1h5.443a1 1 0 0 1 .8.4l2.7 3.6H1ZM9 20V8m0 0-4 4m4-4 4 4" />
    </g>
    <defs>
      <clipPath id="a">
        <path fill="currentColor" d="M0 0h18v18H0z" />
      </clipPath>
    </defs>
  </svg>
);
