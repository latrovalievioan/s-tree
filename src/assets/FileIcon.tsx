import { SVGProps } from "react";

export const FileIcon = (props: SVGProps<SVGSVGElement>) => (
  <svg
    width={20}
    height={20}
    viewBox="0 0 24 24"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    {...props}
  >
    <path
      d="M19 9v8.8c0 1.12 0 1.68-.218 2.108a2 2 0 0 1-.874.874C17.48 21 16.92 21 15.8 21H8.2c-1.12 0-1.68 0-2.108-.218a2 2 0 0 1-.874-.874C5 19.48 5 18.92 5 17.8V6.2c0-1.12 0-1.68.218-2.108a2 2 0 0 1 .874-.874C6.52 3 7.08 3 8.2 3H13m6 6-6-6m6 6h-5a1 1 0 0 1-1-1V3"
      stroke="#fff"
      strokeWidth={2}
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
);
