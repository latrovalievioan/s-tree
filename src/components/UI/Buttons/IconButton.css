.action {
  cursor: pointer;
}
.action>svg:hover {
  color: var(--accent-color);
}

