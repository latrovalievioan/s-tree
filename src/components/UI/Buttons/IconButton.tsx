import './IconButton.css';

type Props = {
  icon: JSX.Element;
  title: string;
  onClick: () => void;
  disabled?: boolean;
};

export const IconButton: React.FC<Props> = ({
  icon,
  title,
  onClick,
  disabled = false,
}) => {
  return (
    <button
      data-testid="icon-button"
      className="action"
      disabled={disabled}
      onClick={onClick}
      title={title}
    >
      {icon}
    </button>
  );
};
