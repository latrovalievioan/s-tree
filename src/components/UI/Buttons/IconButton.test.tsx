import { render, screen, cleanup } from '@testing-library/react';
import { IconButton } from './IconButton';
import { AddFileIcon } from '@/assets/AddFileIcon';

describe(IconButton, () => {
  beforeEach(() => {
    render(<IconButton Icon={AddFileIcon} disabled onClick={() => {}} />);
  });

  afterEach(() => {
    cleanup();
  });

  it('Should render the component', () => {
    const iconButton = screen.getByTestId('icon-button');
    expect(iconButton).toBeInTheDocument();
    expect(iconButton).toBeVisible();
  });

  it('Shoud disable the button when disabled=true', () => {
    const iconButton = screen.getByTestId('icon-button');
    expect(iconButton).toBeDisabled();
  });

  it('Matches snapshot', () => {
    const iconButton = screen.getByTestId('icon-button');

    expect(iconButton).toMatchSnapshot();
  });
});
