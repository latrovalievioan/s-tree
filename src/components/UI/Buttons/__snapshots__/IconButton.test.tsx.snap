// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconButton > Matches snapshot 1`] = `
<button
  class="action"
  data-testid="icon-button"
  disabled=""
  title="Add File"
>
  <svg
    aria-hidden="true"
    fill="none"
    height="32"
    viewBox="0 0 20 20"
    width="32"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7 19H1.933A.97.97 0 0 1 1 18V5.828a2 2 0 0 1 .586-1.414l2.828-2.828A2 2 0 0 1 5.828 1h8.239A.97.97 0 0 1 15 2v4M6 1v4a1 1 0 0 1-1 1H1m11 8h4m-2 2v-4m5 2a5 5 0 1 1-10 0 5 5 0 0 1 10 0Z"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</button>
`;
